### DISCUSSION TITLE
getting unexpected token error while declaring global variable?

### DISCUSSION THREAD

**Discussion by naveennsit** [vercel/next.js#28739] [Help]
I am getting ` **Unexpected token '&'**` error while declaring globle variable .I also tried to stringily it's value but still getting same error why ?

I am doing like this. 

    <Head>
     
    
       {
          <script type="text/javascript">
            var s_pageName2=JSON.stringify('abc:en-us:/lo-form')
          </script>
        }
      </Head> 

here is my whole code
https://codesandbox.io/s/long-bird-tgwcv?file=/pages/index.js

Ia m using nextjs framework . I want to create global variable having value this `abc:en-us:/lo-form`

This is not **codesandbox** error .I am facing same issue on my local machine

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/wZYDR.png

**Comment by oliverloops** [comment]
Hi @naveennsit, the reason why this isn't working it's because you can't use JavaScript  expressions inside JSX (because of it's declarative nature) eg. declaring variable names, use for and while loops or initialize objects, etc. Doesn't matter that everything is wrapped in curly braces Next.js is based  on React and everything inside your script tag is rendered as just JSX. The solution it's to abstract your JavaScript expressions inside another component and call just your variable inside curly braces as this:

```
const Component = () => {
   const s_pageName2 = 'abc:en-us:/lo-form'
   
   return(
      <script>
         {s_pageName2}
      </script>
   )
}
```

By the way if your plan it's to add some information to the html header you can add it using a  _document.js page file provided by Next.js in order to write meta tags or any other header's content and add it globally to your app. you can read more about it here: https://nextjs.org/docs/advanced-features/custom-document

I'll hope this helps.
